Make Escrow.sol pure ERC20 compatible contract

vkomodey opened the pull request:
Some networks(like Polygon Mainnet) have a standard ERC20 implementation of the HMT. This makes it impossible to use HMT native bulk* methods in Escrow contracts. So that's why Escrow's bulkPayOut was modified and now includes the bulk transfer within the contract, instead of executing it on behalf HMT contract.

Existing setups which use previous Escrow implementations should not be broken, since Escrow ABI not changed at all(as well as hmt-escrow interface)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/humanprotocol/hmt-escrow?pullRequest=295) <br/>All committers have signed the CLA.

posix4e commented:
@vkomodey How was this tested?

vkomodey commented:
@posix4e 
We are still testing it. But previous week I and IM performed test job runs on the polygon mainnet with bulk payouts afterward

posix4e commented:
How's this going?

posix4e commented:
@vkomodey is this good to remove WIP

vkomodey commented:
no progress so far, have been busy with other tasks. going to revisit this shortly(tomorrow or day after)

posix4e commented:
1) why is this still a WIP
2) What is blocking merging this?
3) Will this break other things?

vkomodey commented:
@posix4e 
> why is this still a WIP

I'm working on the different concept within this PR. But my work can be implemented in the different PR as well.
> What is blocking merging this?

Considering the previous statement, nothing
> Will this break other things?

No, it won't. Neither the ABI nor the library interface

Let me bump the version. After that we gtg

vkomodey committed:
Make Escrow.sol pure ERC20 compatible contract

Some networks(like Polygon Mainnet) have a standard ERC20 implementation of the HMT. This makes it impossible to use HMT native bulk* methods in Escrow contracts. So that's why Escrow's bulkPayOut was modified and now includes the bulk transfer within the contract, instead of executing it on behalf HMT contract.

vkomodey commented:
cc @rodriguez-facundo

vkomodey commented:
@rodriguez-facundo just a friendly reminder about this pr. is it seems good for you and the team?

rodriguez-facundo reviewed (approved):
yes, looks good
